### ISSUE TITLE
ENH: Add a SpiContainerBootup API ... to enable automatic startup of docker containers (for testing usually)

### ISSUE THREAD

**Issue by rbygrave** [ebean-orm/ebean#1125]
I have got a "docker-commands" project at https://github.com/avaje/docker-commands ... which has support for starting docker containers for Postgres and MySql (and I'll look to extend that to SQL Server, Oracle and ElasticSearch).

The net result is that we will be able to:
- Add a test dependency for ebean-docker-run
- Add a docker-run.properties file that specifies the type and config for containers we wish to run
... and that's it.

Automatically when the Ebean "container" starts it can use "docker-commands" to read the docker-run.properties file and start the docker containers (like Postgres and ElasticSearch) and ensure they are setup with DB user and extensions (like hstore, pgcrypto etc).

In short this will make it very easy to run testing against docker based DB's (which is becoming more important as we use more advanced features like JSON/JSONB, DB Arrays, for update skip locked etc.
